File: 0467-unique-substrings-in-wraparound-string/0467-unique-substrings-in-wraparound-string.java
class Solution {
    public int findSubstringInWraproundString(String s) {
        int n = s.length();
        
        int count[] = new int[26];
        
        int max_len_substring = 1;
        for(int i = 0; i < n; i++) {
            if(i > 0 && (s.charAt(i) - s.charAt(i - 1) == 1 || s.charAt(i - 1) - s.charAt(i) == 25)) {
                max_len_substring++;
            }
            
            else max_len_substring = 1;
            
            int index = s.charAt(i) - 'a';
            count[index] = Math.max(max_len_substring, count[index]);
        }
        
        return Arrays.stream(count).reduce(0, Integer::sum);
    }
}

File: 0467-unique-substrings-in-wraparound-string/0467-unique-substrings-in-wraparound-string.cpp
class Solution {
public:
    int findSubstringInWraproundString(string s) {
        int n = s.length();
        vector<int>ans(26);
        int mx = 1;
        for(int i=0;i<n;i++) {
            if(i>0 && (s[i]-s[i-1]==1 || s[i-1]-s[i]==25)) mx++;
            else mx=1;
            int j = s[i]-'a';
            ans[j] = max(mx, ans[j]);
        }
        return accumulate(ans.begin(), ans.end(), 0);
    }
};